templates/base/src/utils/Logger/logger.js

```diff
@@ -0,0 +1,125 @@
+
+import fetch from 'isomorphic-unfetch';
+
+/**
+ * USAGE: ->
+ * import Logger from '../utils/Logger';
+ *  call as :-
+ * Logger.error({ message: 'Here is the error',
+ *    event: {
+      name: 'event20', id: 657,
+      module: 'modA',
+      type: "load|interaction|redirect" ,
+      data: { a: "hhhh" },
+    },
+    error: {
+      code: 'ABC',
+      message: 'message',
+      operationName: 'testing'
+    },
+    service: {
+      name: 'serviceName',
+      path: 'servicePath',
+    } });
+ */
+
+/**
+ * @param {object} msg --> msg Object
+ * @param {string}  logLevel --> string (info, log, error)
+ * returns an object with log data to be sent in api
+ */
+const getLogObject = ({ msg, logLevel }) => {
+  if (process.browser) {
+    const { message, event: { name: eventName, id, module, type, data }, service: { name: serviceName, path }, error: { code, message: errMsg, operationName } } = msg || {};
+    let location = '';
+    let host = '';
+    let userAgent = '';
+    if (process.browser && window && window.location) {
+      location = window.location.href;
+      host = window.location.host;
+    }
+    if (process.browser && navigator) {
+      userAgent = navigator.userAgent;
+    }
+    return {
+      appName: "universal-react", //Mandatory
+      logLevel,
+      message,
+      browser: { //Mandatory for browser side,
+        location,
+        host,
+        userAgent
+      },
+      userInfo: { //Optional and to be updated by application
+      },
+      event: { //Optional and to be updated by application through log call
+        name: eventName,
```

Comment:
What is the need for creating such a mapping? Why don't we simply spread the event object here?

Reply:
done